src/main/java/io/cdap/plugin/servicenow/apiclient/ServiceNowTableAPIClientImpl.java

```diff
@@ -271,20 +275,37 @@ public Schema fetchTableSchema(String tableName, FailureCollector collector) {
   }
 
   @VisibleForTesting
-  public SchemaResponse parseSchemaResponse(String responseBody) {
+  public MetadataAPISchemaResponse parseSchemaResponse(String responseBody) {
+    return GSON.fromJson(responseBody, MetadataAPISchemaResponse.class);
+  }
+
+  public SchemaResponse parseSchemaResponseWithoutMetadata(String responseBody) {
     return GSON.fromJson(responseBody, SchemaResponse.class);
   }
 
   /**
    * Fetches the table schema from ServiceNow
    *
    * @param tableName ServiceNow table name for which schema is getting fetched
+   * @param valueType Whether to fetch schema for actual value or display value
+   * @param useConnection This flag is added to backward compatibility with 1.1 release,
+   *                      in case of null for useConnection, schema will be string based.
    * @return schema for given ServiceNow table
    * @throws ServiceNowAPIException
    */
-  public Schema fetchTableSchema(String tableName, SourceValueType valueType)
+  public Schema fetchTableSchema(String tableName, SourceValueType valueType, Boolean useConnection)
```

Comment:
Do you see a usecase where we need to get `useConnection` value that would be different from that we are taking in constructor ?